react/components/Popper.tsx

```diff
@@ -0,0 +1,187 @@
+import React, {
+  useRef,
+  useState,
+  useEffect,
+  forwardRef,
+  useCallback,
+  MutableRefObject,
+} from 'react'
+import {
+  createPopper,
+  VirtualElement,
+  ModifierArguments,
+  State as PopperState,
+  Instance as PopperInstance,
+  Placement as PopperPlacementType,
+} from '@popperjs/core'
+import { useCssHandles } from 'vtex.css-handles'
+
+import Portal from './Portal'
+import styles from '../styles.css'
+import setRef from '../modules/setRef'
+import useForkRef from '../modules/useForkRef'
+
+interface Props
+  extends React.DetailedHTMLProps<
+    React.HTMLAttributes<HTMLDivElement>,
+    HTMLDivElement
+  > {
+  anchorEl?:
+    | null
+    | VirtualElement
+    | (() => VirtualElement)
+    | React.RefObject<HTMLElement>
+  children: React.ReactNode
+  open?: boolean
+  placement?: PopperPlacementType
+  transition?: boolean
+  role?: string
+}
+
+function getAnchorEl(anchorEl: React.ReactNode) {
+  return typeof anchorEl === 'function' ? anchorEl() : anchorEl
+}
+
+const CSS_HANDLES = ['popper']
+
+const Popper = forwardRef(function Popper(props: Props, ref) {
+  const {
+    role,
+    anchorEl,
+    children,
+    open = false,
+    transition = false,
+    placement: initialPlacement = 'bottom',
+    ...rest
+  } = props
+
+  const tooltipRef = useRef<HTMLElement>(null)
+  const ownRef = useForkRef(tooltipRef, ref)
+  const popperRef: MutableRefObject<PopperInstance | null> = useRef<
+    PopperInstance
+  >(null)
+  const [placement, setPlacement] = useState(initialPlacement)
+  const [exited, setExited] = useState(true)
+  const handles = useCssHandles(CSS_HANDLES)
+
+  // It doesn't have a dependency array because it should update
+  // after every render of react
+  useEffect(() => {
+    if (popperRef.current) {
+      popperRef.current.update()
+    }
+  })
```

Comment:
Why not am empty dependency array then?

Reply:
empty dependency array will trigger only after the first render